addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -428,25 +428,21 @@ private void sendPulseCommand(ChannelUID channelUID, ChannelParams params, WSRes
             // set resource to ON
             logger.debug("Update resource value (inverted output={}): {}", params.isInverted(), valOn);
             if (updateResource(valOn)) {
-                scheduler.submit(() -> {
-                    // sleep a while
-                    try {
-                        logger.debug("Sleeping: {}ms", pulseWidth);
-                        Thread.sleep(pulseWidth);
-                    } catch (InterruptedException e) {
-                        // do nothing
-                    }
-                    // set resource back to OFF
-
-                    logger.debug("Update resource value (inverted output={}): {}", params.isInverted(), valOff);
-                    try {
-                        if (!updateResource(valOff)) {
-                            logger.warn("Channel {} update to resource '{}' failed.", channelUID, valOff);
+                logger.debug("Sleeping: {}ms", pulseWidth);
+                scheduler.schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        // set resource back to OFF
+                        logger.debug("Update resource value (inverted output={}): {}", params.isInverted(), valOff);
+                        try {
+                            if (!updateResource(valOff)) {
+                                logger.warn("Channel {} update to resource '{}' failed.", channelUID, valOff);
+                            }
+                        } catch (IhcExecption e) {
+                            logger.error("Can't update channel '{}' value, cause ", channelUID, e.getMessage(), e);
```

Comment:
Can you change log levels from error to warn in a follow up PR? That would be really great.

Reply:
Yes I can do it, but I made quick search to openhab2 addons project and there seems to be almost 1500 error levels used over 100 bindings 😃

Reply:
Yup. They are all wrong. But good point, need to be automatically replaced.
See also my PR on the contribution page: https://github.com/openhab/openhab-docs/pull/933